packages/pwa-kit-react-sdk/src/ssr/universal/components/redirect-with-status/index.jsx

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React from 'react'
+import {Redirect, Route} from 'react-router-dom'
+import PropTypes from 'prop-types'
+
+/**
+ * The `RedirectWithStatus` HOC component is used to specify a different status code when redirecting via
+ * the Redirect component.
+ * The default redirect behavior when this component is not used is to set a 302 status.
+ *
+ * @param {number} status - The HTTP status code. Defaults to 302 if not specified
+ * @param {string} to - The redirect's target path 
+ */
+export const RedirectWithStatus = ({status = 302, ...props}) => {
+    return (
+        <Route
+            render={({staticContext}) => {
+                if (staticContext) staticContext.status = status
+                return <Redirect {...props} />
+            }}
+        />
+    )
+}
```

Comment:
Do you mind trying the following implementation:

```
import React from 'react'
import {Redirect, withRouter} from 'react-router-dom'

const RedirectWithStatus = ({statusCode = 302, staticContext, ...props}) => {
  // Handle server-side rendering
  if (staticContext) {
    staticContext.statusCode = statusCode
  }

  return <Redirect {...props} />
}

export default withRouter(RedirectWithStatus)
```

This solution is less hacky than what we had previously. Let me know how that works out.

Reply:
This implementation works and is now what this PR is using!